src/Executables/Auth/PacketHandlers/LoginRequestHandler.cs

```diff
@@ -59,16 +83,25 @@ public async Task ExecuteAsync(AuthPacketContext<LoginRequest> ctx, Cancellation
         }
         catch (Exception e)
         {
-            _logger.LogWarning("Failed to verify password for account {Username}: {Message}", ctx.Packet.Username, e.Message);
-            status = "WRONGPWD";
+            _logger.LogWarning("Failed to verify password for account {Username}: {Message}", ctx.Packet.Username,
+                e.Message);
+            status = LoginFailedBecause.InvalidCredentials.AsString(EnumFormat.EnumMemberValue);
         }
 
+        // Check if the account is already logged in
+        var isAlreadyLoggedIn = await CheckExistingConnectionOfAsync(account);
+        if (isAlreadyLoggedIn)
+        {
+            status = await DecideAlreadyLoggedInStatusAsync(account);
+        }
+
+
         // If the status is not empty send a failed login response to the client
         if (status != "")
         {
             ctx.Connection.Send(new LoginFailed
             {
-                Status = status
+                Status = status ?? "UNKNOWN"
```

Comment:
use enum here

Reply:
This one is a workaround of `Enums.NET`'s `AsString` method. There was a code smell and possible `null` reference assignment. If there is a best practice of that I didn't fimiliar with, please let me know! :)

Reply:
Feel free to suppress this null warning :)